src/relion/dbmodel/modeltables.py

```diff
@@ -113,6 +99,7 @@ def add_row(self, row):
                                 except TypeError:
                                     self._tab[c][index].append(row.get(c))
                                     self._tab[c][index] = list(set(self._tab[c][index]))
+                                    # should these be sets by default?
```

Comment:
In the `try` block you also do extend/set/list, so that makes me wonder whether `_tab[][]` elements should just be `set`s?

Reply:
Yes, I think that makes sense. I put in this `append` setting so that I could "connect" multiple `ParticleClassification` table rows to the same `CryoemInitialModel` (what the `ParticleClassification_has_CryoemInitialModel` does in the actual database schema). In that case you don't want repeat values of the same `ParticleClassification` table ID.

Reply:
do you care about order though? If we do, then we could still use an ordered set.

Reply:
I don't think we do care. On the ISPyB service side we just need to know which `ParticleClassification` rows should point to a given `CryoemInitialModel` row. The order shouldn't matter